sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_asset_utils.py

```diff
@@ -652,6 +652,33 @@ def _create_or_update_autoincrement(
     return result
```

Comment:
Your new method is really similar to the code from `create_or_update_autoincrement` above. Could you please replace the first section of that method with your new method as a helper?

```suggestion
def _create_or_update_autoincrement(
    name: str,
    body: Any,
    version_operation: Any,
    container_operation: Any,
    resource_group_name: str,
    workspace_name: str,
    registry_name: str = None,
    **kwargs,
) -> Any:
    version = _get_next_version_from_container(..)

    result = version_operation.create_or_update(
        name=name,
        version=version,
        resource_group_name=resource_group_name,
        workspace_name=workspace_name,
        body=body,
        **kwargs,
    )
    return result
```

Reply:
As we chatted, the new method will replace the old one when we have model and environment have next_version implemented also may be in next few weeks. For now, the old method is there for back compatability.